app/controllers/api/proxy_rules_controller.rb

```diff
@@ -0,0 +1,64 @@
+class Api::ProxyRulesController < Api::BaseController
+  before_action :find_service
+  before_action :find_proxy
+
+  activate_menu :serviceadmin, :integration, :mapping_rules
+
+  sublayout 'api/service'
+
+  def index
+    @proxy_rules = @proxy.proxy_rules.includes(:metric).ordered.paginate(page: params[:page])
+  end
+
+  def new
+    next_position = @proxy.proxy_rules.maximum(:position) || 0
```

Comment:
The variable name `next_position` is a bit misleading, as it's actually `last_position`, and next position would be last + 1.